packages/story-editor/src/components/library/panes/shopping/shoppingPane.js

```diff
@@ -13,81 +13,204 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
 /**
  * External dependencies
  */
+import PropTypes from 'prop-types';
 import { useFeature } from 'flagged';
 import styled from 'styled-components';
-import { useCallback, useState } from '@googleforcreators/react';
-import { __ } from '@googleforcreators/i18n';
 import {
-  Button,
-  BUTTON_VARIANTS,
-  BUTTON_TYPES,
-  BUTTON_SIZES,
+  useCallback,
+  useState,
+  useEffect,
+  useDebouncedCallback,
+} from '@googleforcreators/react';
+import { __, sprintf } from '@googleforcreators/i18n';
+import {
   Text,
   THEME_CONSTANTS,
+  SearchInput,
+  useLiveRegion,
 } from '@googleforcreators/design-system';
 
 /**
  * Internal dependencies
  */
+import { Section } from '../../common';
+import { useAPI } from '../../../../app';
 import { Row } from '../../../form';
-import useLibrary from '../../useLibrary';
 import { Pane } from '../shared';
+import { useStory } from '../../../../app/story';
+import useLibrary from '../../useLibrary';
 import paneId from './paneId';
-import ProductDropdown from './productDropdown';
+import ProductList from './productList';
 
 const HelperText = styled(Text).attrs({
   size: THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL,
 })`
   color: ${({ theme }) => theme.colors.fg.secondary};
 `;
+function NoProductsFound({ searchTerm }) {
+  if (searchTerm?.length >= 1) {
+    return <HelperText>{__('No products found.', 'web-stories')}</HelperText>;
+  }
+  return null;
+}
+
+NoProductsFound.propTypes = {
+  searchTerm: PropTypes.string,
+};
 
 function ShoppingPane(props) {
-  const isEnabled = useFeature('shoppingIntegration');
-  const [product, setProduct] = useState(null);
+  const isShoppingIntegrationEnabled = useFeature('shoppingIntegration');
+  const speak = useLiveRegion('assertive');
+  const [searchTerm, setSearchTerm] = useState('');
+  const [isMenuFocused, setIsMenuFocused] = useState(false);
+  const [products, setProducts] = useState([]);
+  const {
+    actions: { getProducts },
+  } = useAPI();
+
+  const { currentPageProducts } = useStory(({ state: { currentPage } }) => {
+    return {
+      currentPageProducts: currentPage?.elements
+        ?.filter(({ type }) => type === 'product')
+        .map(({ id, product }) => ({
+          elementId: id,
+          productId: product?.productId,
+        })),
+    };
+  });
+
+  const getProductsByQuery = useCallback(
+    async (value = '') => {
+      try {
+        setProducts(await getProducts(value));
+      } catch (err) {
+        setProducts([]);
+      }
+    },
+    [getProducts]
+  );
+
+  const debouncedProductsQuery = useDebouncedCallback((value) => {
+    getProductsByQuery(value);
+  }, 300);
+
+  const onSearch = useCallback(
+    (evt) => {
+      const value = evt.target.value;
+      if (value !== searchTerm) {
+        setSearchTerm(value);
+        debouncedProductsQuery(value);
```

Comment:
Wondering if search should be trigger by an useEffect? So this should only set search state and a debounced use effect should do the search. Might make the logic a little cleaner.

Reply:
Updated and yes seems a little cleaner.